components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/util/APIResourceManagementUtil.java

```diff
@@ -87,22 +90,63 @@ public static void addSystemAPIs() {
             Map<String, APIResource> configs = APIResourceManagementConfigBuilder.getInstance()
                     .getAPIResourceMgtConfigurations();
+            Map<String, APIResource> duplicateConfigs = APIResourceManagementConfigBuilder.getInstance()
+                    .getDuplicateAPIResourceConfigs();
             if (!isSystemAPIExist(tenantDomain)) {
                 LOG.debug("Registering system API resources in the server.");
                 registerAPIResources(new ArrayList<>(configs.values()), tenantDomain);
             } else {
                 LOG.debug("System APIs are already registered in the server. Applying the latest configurations.");
+                // Remove the existing system APIs from the configs.
+                // Existing system APIs will be evaluated using the identifier.
+                HashMap<String, APIResource> tempConfigs = new HashMap<>(configs);
                 List<APIResource> systemAPIs = getSystemAPIs(tenantDomain);
                 for (APIResource systemAPI : systemAPIs) {
-                    if (configs.containsKey(systemAPI.getIdentifier())) {
-                        configs.remove(systemAPI.getIdentifier());
+                    if (tempConfigs.containsKey(systemAPI.getIdentifier())) {
+                        tempConfigs.remove(systemAPI.getIdentifier());
                     } else {
                         String apiId = APIResourceManagerImpl.getInstance().getAPIResourceByIdentifier(
                                 systemAPI.getIdentifier(), tenantDomain).getId();
                         APIResourceManagerImpl.getInstance().deleteAPIResourceById(apiId, tenantDomain);
                     }
                 }
-                registerAPIResources(new ArrayList<>(configs.values()), tenantDomain);
+                // Register the new system APIs.
+                registerAPIResources(new ArrayList<>(tempConfigs.values()), tenantDomain);
+
+                // Handle duplicate system APIs.
+                for (APIResource oldAPIResource : duplicateConfigs.values()) {
+                    // Get the existing API resource from the DB.
+                    APIResource apiResourceFromDB = APIResourceManagerImpl.getInstance().getAPIResourceByIdentifier(
+                            oldAPIResource.getIdentifier(), tenantDomain);
+                    // Get the updated API resource from the configs.
+                    APIResource updatedAPIResource = configs.get(oldAPIResource.getIdentifier());
+                    // Get the scopes which are not in the existing API resource.
+                    List<Scope> addedScopes = updatedAPIResource.getScopes().stream()
+                            .filter(scope1 -> apiResourceFromDB.getScopes().stream()
+                                    .noneMatch(scope2 -> scope2.getName().equals(scope1.getName())))
+                            .collect(Collectors.toList());
+                    if (addedScopes.isEmpty()) {
```

Comment:
Here we are not allowing removing or updating scopes of a given system API. Consider it based on the usecases

Reply:
Since the removal of the system API scopes are not required, this is not implemented.